libvips/foreign/vips2tiff.c

```diff
@@ -1300,6 +1315,82 @@ eightbit2onebit( Wtiff *wtiff, VipsPel *q, VipsPel *p, int n )
 		*q++ = bits << (8 - (x & 0x7));
 }
 
+/* Pack 8 bit VIPS to 2 bit TIFF.
+ */
+static void
+eightbit2twobit( Wtiff *wtiff, VipsPel *q, VipsPel *p, int n )
+{
+	int x;
+	VipsPel bits;
+
+	bits = 0;
+	if( !wtiff->miniswhite ) { //avoid unnecessary branches
+		for( x = 0; x < n; x++ ) {
+			bits <<= 2;
+			bits |= p[x] >> 6;
```

Comment:
You could do this as:

```C
bits |= (p[x] >> 6) ^ mask;
```

where `mask` is `wtiff->miniswhite ? 3 : 0` somewhere outside the loop. It would save the loop copy-paste.